### PR TITLE
Feature/albums

### PR DESCRIPTION
**Pull request by larkinds**
Fully functioning albums sidebar with tracks that play/add-to-queue

### CODE REVIEW

File: src/albums.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  }
+
+  async getTracks(id: string, spotify: SpotifyWebApi) {
+    // album.tracks.items[0].name
+    let tracks = await spotify.getAlbumTracks(id);
+    tracks.body.items[0].name;
+    return tracks.body.items.map((track: any) => new AlbumsAndTracks(track));
+  }
+}
```
Comment (jkmartindale) [inline src/albums.ts]: Gimme that end of file line break :'(
```suggestion
}

```

File: src/albums.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      this.parent = parent;
+
+      this.description = `${data.name} | ${this.artists}`;
+      this.contextValue = 'albumTrack';
+    }
+
+  }
+
+  async getTracks(id: string, spotify: SpotifyWebApi) {
+    // album.tracks.items[0].name
```
Comment (jkmartindale) [inline src/albums.ts]: ?

File: src/albums.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+export class AlbumsAndTracks extends vscode.TreeItem {
+  name: string;
+  artists: string;
+  uri: string;
+  children: Promise<AlbumsAndTracks[]> | undefined;
+  parent: AlbumsAndTracks | undefined;
+
+  constructor (public readonly data: any, spotify?: SpotifyWebApi, parent?: AlbumsAndTracks) {
+    super(data);
```
Comment (jkmartindale) [inline src/albums.ts]:
```suggestion
    super(data.name ?? data.album.name);
```

File: src/albums.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      this.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+      this.contextValue = 'album';
+    } else {
+      this.name = data.name;
+      this.artists = data.artists.map((artist: any) => artist.name).join(', ');
+      this.uri = data.uri;
+      this.children = undefined;
+      this.parent = parent;
+
+      this.description = `${data.name} | ${this.artists}`;
```
Comment (jkmartindale) [inline src/albums.ts]:
```suggestion
      this.description = this.artists;
```

**Review by jkmartindale — CHANGES_REQUESTED** [review]
Just wanna see the track labels fixed like with playlists

Since you branched off `feature/playlist` instead of `staging` (totally understand how easy it is to accidentally branch off another feature branch), we'll need to wait until #30 gets merged in, and then very carefully resolve merge conflicts coming from this branch's copy of the work in `feature/playlist`. Should probably be merged in with **Squash and Merge**.

File: src/albums.ts (hunk 2 above)
Comment (larkinds) [reply src/albums.ts]: Yeahhhhhh 😋

**Pull request merged**